scanner/e2etests/image_test.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/stackrox/rox/scanner/client"
+	"github.com/stackrox/rox/scanner/indexer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Vulnerability describes a vulnerability in a TestCase.
+type Vulnerability struct {
+	Name        string                 `json:"Name"`
+	Description string                 `json:"Description"`
+	Link        string                 `json:"Link"`
+	Severity    string                 `json:"Severity"`
+	Metadata    map[string]interface{} `json:"Metadata"`
+	FixedBy     string                 `json:"FixedBy"`
+}
+
+// Vulnerability describes a feature in a TestCase.
+type Feature struct {
+	Name            string          `json:"Name"`
+	NamespaceName   string          `json:"NamespaceName"`
+	VersionFormat   string          `json:"VersionFormat"`
+	Version         string          `json:"Version"`
+	Vulnerabilities []Vulnerability `json:"Vulnerabilities"`
+	AddedBy         string          `json:"AddedBy"`
+	Location        string          `json:"Location"`
+	FixedBy         string          `json:"FixedBy"`
+}
+
+// TestWant describes the information wanted from the scan results in a TestCase.
+type TestWant struct {
+	Namespace string    `json:"namespace"`
+	Source    string    `json:"source"`
+	Features  []Feature `json:"expected_features"`
+}
+
+// TestArgs describes the arguments (i.e. test input) necessary to exercise the test case.
+type TestArgs struct {
+	Image    string `json:"image"`
+	Registry string `json:"registry"`
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// TestCase represents on test case or scenario.
+type TestCase struct {
+	TestArgs
+	TestWant
+	// Test flags.
+	OnlyCheckSpecifiedVulns  bool `json:"only_check_specified_vulns"`
+	UncertifiedRhel          bool `json:"uncertified_rhel"`
+	CheckProvidedExecutables bool `json:"check_provided_executables"`
+}
+
+// TestImage is the E2E test for container image scans.
+func TestImage(t *testing.T) {
+	ctx := context.Background()
+	c, err := client.NewGRPCScannerClient(ctx,
+		client.WithAddress(":8443"),
+		client.WithoutTLSVerify)
+	require.NoError(t, err)
+
+	testCases, err := loadTestCases("testdata/image_tests.json")
+	require.NoError(t, err)
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s/%s", tc.Namespace, tc.Image), func(t *testing.T) {
+			ref, err := name.ParseReference(tc.Image)
+			require.NoError(t, err)
+
+			auth := tc.authConfig()
+			d, err := indexer.GetDigestFromReference(ref, auth)
+			require.NoError(t, err)
+
+			vr, err := c.IndexAndScanImage(ctx, d, auth)
+			require.NoError(t, err)
+
+			expected := &tc.TestWant
+			converted := tc.convertReport(vr)
+			defer func() {
+				if t.Failed() && vr != nil {
+					tc.logReport(t, vr)
+				}
+			}()
+			assert.Equal(t, expected, converted,
+				"The vulnerability report converted did not contain the expected values.")
+		})
+	}
+}
+
+func loadTestCases(s string) ([]*TestCase, error) {
```

Comment:
nit: you can do `mustLoadTestCases(t *testing.T, s string) []*TestCase` and then `t.Require` or `t.Fatal` if this fails. Remove the need to check errors from the caller

Reply:
I wanted to make this function generic and not tied to `testing`. Checking the error is not a burden and simplifies the function signature.